tests/test_templates.py

```diff
@@ -18,24 +19,31 @@ class TestTemplateBase(object):
     project_class = signac.Project
     env = None
 
-    def env_name(self):
+    def _env_name(self):
         name = '{}.{}'.format(self.env.__module__, self.env.__name__)
         return '.'.join(name.split('.')[1:])
 
-    def get_TestEnvironment(self):
+    def find_envs():
+        for name, env in flow.environment.ComputeEnvironment.registry.items():
+            if env.__module__.startswith('flow.environments'):
+                yield env
+
+    @pytest.mark.parametrize('env', find_envs())
+    def test_env(self, env):
```

Comment:
This should be a module-level function now, there is no need to have a test class anymore.

Reply:
If I don't use self, this error comes up.
In test_env: function uses no argument 'env'

Reply:
What I can do is, remove the class and just let test_templates.py to be a set of functions.

Reply:
Yes, exactly, that's what I meant by "not have a test class".

Reply:
Yes I just read it carefully, having a bad day I guess. Sorry about that.